### ISSUE TITLE
Rule Proposal: `vue/require-default-prop`

### ISSUE THREAD

**Issue by armano2** [vuejs/eslint-plugin-vue#122]
## Please describe what the rule should do:

This rule aims to ensure that any non-required `prop` declaration of a component has a corresponding `default` value.

## What category of rule is this? (place an "X" next to just one item)
`[ ]` Enforces code style
`[x]` Warns about a potential error
`[ ]` Suggests an alternate way of doing something
`[ ]` Other (please specify:)

Provide 2-3 code examples that this rule will warn about:

```js
export default {
  props: {
     foo: {
         type: String
     }
  }
}
```

```js
export default {
  props: {
     foo: {
         type: Array,
         required: false
     }
  }
}
```

based on conversation from: https://github.com/vuejs/eslint-plugin-vue/issues/117#issuecomment-319317832

**Comment by michalsnik** [comment]
I'll pick this

**Comment by dsl101** [comment]
Just trying to understand this rule. Why does this code:
```
props: {
  id: 0
}
```
generate an error? Isn't a literal value for a prop the same as a default value?
